Server/ReasnAPI/ReasnAPI/Services/ParameterService.cs

```diff
@@ -1,16 +1,85 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
 
-namespace ReasnAPI.Services {
-    public class ParameterService (ReasnContext context){
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+namespace ReasnAPI.Services;
+public class ParameterService (ReasnContext context)
+{
+    public ParameterDto CreateParameter(ParameterDto parameterDto)
+    {
+        var parameter = context.Parameters.FirstOrDefault(r => r.Key == parameterDto.Key);
+        if (parameter != null)
+        {
+            return null;
+        }
+
+        var newParameter = new Parameter
+        {
+            Key = parameterDto.Key,
+            Value = parameterDto.Value
+        };
+        context.Parameters.Add(newParameter);
+        context.SaveChanges();
+        return parameterDto;
+    }
+
+    public ParameterDto UpdateParameter(int parameterId,ParameterDto parameterDto)
+    {
+        var parameter = context.Parameters.FirstOrDefault(r => r.Id == parameterId);
```

Comment:
For now, I'm checking if any event is associated with the parameter. If there is, I am returning the function.

Reply:
Okay, but in future I assume there will be an option to update parameters associated with events, right? It might be done as I proposed. For now, this method seems to be unusable, we can only change parameter if it's not assigned.